apps/studio/src/pages/sites/[siteId]/folders/[folderId]/index.tsx

```diff
@@ -1,34 +1,60 @@
+import { Suspense, useEffect } from "react"
 import {
   Box,
   BreadcrumbItem,
   BreadcrumbLink,
   Flex,
+  FormControl,
+  FormErrorMessage,
+  FormHelperText,
+  FormLabel,
   HStack,
+  Icon,
+  Input,
   MenuButton,
   MenuList,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
   Portal,
+  Skeleton,
   Spacer,
   Text,
   useDisclosure,
   VStack,
 } from "@chakra-ui/react"
-import { Breadcrumb, Button, Menu } from "@opengovsg/design-system-react"
-import { uniqueId } from "lodash"
-import { BiData, BiFileBlank, BiFolder } from "react-icons/bi"
+import {
+  Breadcrumb,
+  Button,
+  Menu,
+  useToast,
+} from "@opengovsg/design-system-react"
+import { BiData, BiFileBlank, BiFolder, BiLink } from "react-icons/bi"
 import { z } from "zod"
 
 import { MenuItem } from "~/components/Menu"
 import { ResourceTable } from "~/features/dashboard/components/ResourceTable"
 import { CreateFolderModal } from "~/features/editing-experience/components/CreateFolderModal"
 import { CreatePageModal } from "~/features/editing-experience/components/CreatePageModal"
+import { generateResourceUrl } from "~/features/editing-experience/components/utils"
 import { useQueryParse } from "~/hooks/useQueryParse"
+import { useZodForm } from "~/lib/form"
 import { type NextPageWithLayout } from "~/lib/types"
+import {
+  editFolderSchema,
+  MAX_FOLDER_PERMALINK_LENGTH,
+  MAX_FOLDER_TITLE_LENGTH,
+} from "~/schemas/folder"
 import { AdminCmsSidebarLayout } from "~/templates/layouts/AdminCmsSidebarLayout"
 import { trpc } from "~/utils/trpc"
 
 const folderPageSchema = z.object({
-  siteId: z.coerce.number(),
-  folderId: z.coerce.number(),
+  siteId: z.string(),
```

Comment:
wa actually i changed it to `string` here - i agree with the reasoning you gave that the `bigint` is just to autoincrement automatically for id, but on the frontend we can just treat as a string since we'll never interact with it.

i think changing to `string` here is ok - i'll have a PR to change number -> string later on wdyt?